### PR TITLE
[M651] New notification model and endpoint

### PR DESCRIPTION
**Pull request by gridcell**
Card: [M651](https://trello.com/c/Mf9DIlIS/651-bell-card-models-and-api-endpoints)

### CODE REVIEW

**Comment by gridcell** [comment]
[Bell card models and api endpoints](https://trello.com/c/Mf9DIlIS/651-bell-card-models-and-api-endpoints)

File: src/api/models/mermaid.py [hunk 1]
```diff
@@ -1738,3 +1738,24 @@ class AuditRecord(JSONMixin):
...
+    ERROR = "error"
+    
+    STATUSES = (
+        (INFO, INFO),
+        (WARNING, WARNING),
+        (ERROR, ERROR),
+    )
+
+    title = models.CharField(max_length=200)
+    status = models.CharField(max_length=10, choices=STATUSES)
```
Comment (saanobhaai) [inline src/api/models/mermaid.py:1755]: Just making sure we don't want to store ordinal values, to allow for ranking, like we do for project status?

File: src/api/models/mermaid.py [hunk 2]
```diff
@@ -1738,3 +1738,24 @@ class AuditRecord(JSONMixin):
...
+    description = models.TextField(null=True, blank=True)
```
Comment (saanobhaai) [inline src/api/models/mermaid.py:1756]: Just wondering how this would translate into the types of cards shown on https://trello.com/c/mWx3Nbhf/357-bell-notifications-bell-cards-for-discussion

- the first example I guess we're not doing in the bell anymore
- the second could use this field; this assumes that the app will detect a version upgrade and refresh itself though. Do we have a card for that?
- the third looks like an amalgamation of several `Notification` instances; I'm not sure this is quite right. But if it was, how would we generate "go to project"?

Basically, I'm wondering if the frontend will need some kind of `context` object; maybe the `data` field on `BaseModel` will serve this purpose.

File: src/api/models/mermaid.py [hunk 3]
```diff
@@ -1738,3 +1738,24 @@ class AuditRecord(JSONMixin):
     event_by = models.ForeignKey("Profile", on_delete=models.SET_NULL, null=True, blank=True)
     model = models.CharField(max_length=100)
     record_id = models.UUIDField(db_index=True)
+
+
+class Notification(BaseModel):
```
Comment (saanobhaai) [inline src/api/models/mermaid.py:1743]: So `created_on` in `BaseModel` would be used for the datestamp, right?

**Review by saanobhaai — CHANGES_REQUESTED** [review]

File: src/api/models/mermaid.py (hunk 3 above)
Comment (gridcell) [reply src/api/models/mermaid.py:1743]: yeah

File: src/api/models/mermaid.py (hunk 2 above)
Comment (gridcell) [reply src/api/models/mermaid.py:1756]: Think of description as storing the same content as a WYSIWYG.  The description will support "markup"`
